core/startos/src/service/service_effect_handler.rs

```diff
@@ -336,9 +321,52 @@ async fn clear_network_interfaces(context: EffectContext, _: Empty) -> Result<Va
 }
 async fn export_service_interface(
     context: EffectContext,
-    data: ExportServiceInterfaceParams,
-) -> Result<Value, Error> {
-    todo!()
+    ExportServiceInterfaceParams { id, name, description, has_primary, disabled, masked, address_info, r#type, host_kind, hostnames }: ExportServiceInterfaceParams,
+) -> Result<(), Error> {
+    let context = context.deref()?;
+    let package_id = context.id.clone();
+    let svc_interface_id = ServiceInterfaceId::from(Id::try_from(id)?);
+    let host_id = HostId::from(Id::try_from(address_info.host_id.clone())?);
+    
+    let service_interface = ServiceInterface{
+        id: svc_interface_id.clone(),
+        name,
+        description,
+        has_primary,
+        disabled,
+        masked,
+        address_info,
+        interface_type: r#type,
+    };
+    let host_info = ExportedHostInfo {
+        id: host_id,
+        kind: host_kind,
+        hostnames,
+    };
+    let svc_interface_with_host_info = ServiceInterfaceWithHostInfo {
+        service_interface,
+        host_info,
+    };
+    
+    context
+        .ctx
+        .db
+        .mutate(|db| {
+            let model = db
+                .as_public_mut()
+                .as_package_data_mut()
+                .as_idx_mut(&package_id)
+                .or_not_found(&package_id)?
+                .as_service_interfaces_mut();
+            let mut value = model.de()?;
+            value
+                .insert(ServiceInterfaceId::from(svc_interface_id), svc_interface_with_host_info)
+                .map(|_| ())
+                .unwrap_or_default();
+            model.ser(&value)
```

Comment:
use .insert on the model itself

Reply:
https://github.com/Start9Labs/start-os/blob/next/minor/core/startos/src/db/prelude.rs#L250-L264